src/App.tsx

```diff
@@ -8,7 +8,57 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [searchType, setSearchType] = useState('all');
+  const [query, setQuery] = useState('');
+
+  const formattedQuery = query.trim().toLowerCase();
+
+  const filter = (type: string) => {
+    const findQuery = (queryToSearch: string, todos: Todo[]) => {
+      return todos.filter((todo) => (
+        todo.title.toLowerCase().includes(queryToSearch)
+      ));
+    };
+
+    switch (type) {
```

Comment:
switch as well, and inside filter add condition for 'all', 'active' and 'completed'